pkg/webhook/kubernetes/tls.go

```diff
@@ -54,14 +56,32 @@ func SetupSecureService(client crclient.Client, ctx context.Context, namespace s
 			if err != nil {
 				return err
 			}
+			log.Info("Warning: the webhook server cert in use is not suitable for production. If you want to use this in production please set up certs with a certificate manager")
+			return nil
 		}
 	}
 
-	err = service.CreateOrUpdateSecureService(client, ctx, namespace, map[string]string{})
+	// Check if the secret found at server.WebhookServerTLSSecretName is a cert manager secret. If it is then we can
+	// just inject the ca automatically using cert manager annotations
+	secureServiceAnnotations := make(map[string]string)
+	if isCertManagerSecret(devworkspaceSecret.GetAnnotations()) {
+		secureServiceAnnotations["cert-manager.io/inject-ca-from-secret"] = server.WebhookServerTLSSecretName
```

Comment:
Could you elaborate more on why we need it? I assume that this annotation should be put info webhook configurations if we don't want to propagate it from WebhookServerTLSSecretName manually, as we currently do on webhook server side.

Reply:
I think it's redundant here. Everything that we may be interested to do with `WebhookServerTLSSecretName` secret - check that needed keys are in place, cert-manager provides the following:
![Screenshot_20201013_131835](https://user-images.githubusercontent.com/5887312/95848533-d8c3ef00-0d56-11eb-8065-cf8771ab58c5.png)
but we require only tls.key and tls.crt